### PR TITLE
add new method ar.addUIView (only interactive on android at the moment)

### PR DESCRIPTION
**Pull request by nickolanack**
```js

// Example
ar.addSphere({
	radius:0.25,
	position:{x:0, y:0, z:-1}	
}).then(function(shape){

	ar.addUIView({
		position:{x:0, y:.4, z:0},
		parentNode:shape,
		view:nativescriptView
	})
});
```

this should work if you create ui components programatically or if you use getViewById

### CODE REVIEW

File: src/nodes/android/aruiview.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            if (view) {
+
+                if (view.parent) {
+                    view.parent.removeChild(view);
+                }
+
+
+
+                if (!view.android) {
```
Comment (nickolanack) [inline src/nodes/android/aruiview.ts:33]: @EddyVerbruggen  this seems to allow nativescript ui components generated programatically to work with css, but is there a better way?

**Comment by nickolanack** [comment]
btw Im getting an iPhone today... so I'll work on ios implementation

**Comment by EddyVerbruggen** [comment]
Really excited for this one! It will open up a lot of cool possibilities for sure.

And we need it with #41; I thought it would be useful to implement one or more of the Android samples to see which tools/features we're missing in the plugin to be able to fully recreate it on both platforms.

**Comment by nickolanack** [comment]
I've implemented a version of addUIView for ios here: https://github.com/EddyVerbruggen/nativescript-ar/compare/master...nickolanack:iosUIView2 although it does not work as I had hoped (yet)... -  it is rendering a UIImage of of the UIView and using that as plane material

**Comment by EddyVerbruggen** [comment]
Hi @nickolanack, I've added a quick demo on Android for this feature and had to change a parent.removeView reference to make it work. Pretty stoked for this one! I hope something similar will be possible on iOS.

To check out the demo, go to src and run `npm run demo.android`. Then detect a plane and tap it. That will (among other things) add a yellow sphere to the scene, and our custom UI will be attached to said sphere. Result:

Screenshot_20190817-210554_demo

**Comment by nickolanack** [comment]
Hey @EddyVerbruggen, i was hoping to try to run your ^^^demo on ios. I tried to just use 
`npm run demo.ios`, but i got:
`CocoaPods could not find compatible versions for pod "MaterialComponents/Tabs"`
wondering if you have a quick solution for me? if not - no worries

**Comment by EddyVerbruggen** [comment]
@nickolanack Yes, please run `pod repo update` on you mac, then try again.

**Comment by nickolanack** [comment]
yes! i was just trying that. thanks!

**Comment by EddyVerbruggen** [comment]
Btw, I've been trying to make the iOS version interactive (the `UIImage` approach worked fine), and it worked, but the dimensions of the created view were all weirdly stretched etc so I put it on hold for now. I could push it to your fork for you to play with, but it's not too different from what you already have.

**Comment by nickolanack** [comment]
sure i'd love to see what you came up with. I've been trying some alternate approaches as well... 
I'd also like to add another node type (lol) ar.addImage({image:ImageSource|string}) which would simplify adding billboard type renderables

**Comment by EddyVerbruggen** [comment]
@nickolanack That image node type is a great idea, and you've already got that pretty much working in the fork.

Let me try and push my (half-assed) approach to this fork tomorrow.

**Comment by nickolanack** [comment]
~~I've sort of figured out the scaling issue in ios here: https://github.com/nickolanack/nativescript-ar/blob/b10217543cb9e1d56218220e0c7b42954d6a9982/src/nodes/ios/aruiview.ts#L48-L62
but have a small offset to deal with...~~

**Comment by nickolanack** [comment]
IMG_0011

**Comment by nickolanack** [comment]
I got it working pretty well for ios  (at least for your demo). It doesn't work well for programatically generated layouts though...

**Comment by EddyVerbruggen** [comment]
@nickolanack It's much better than what I came up with!

**Comment by EddyVerbruggen** [comment]
I've merged this in order to prevent further merge conflicts. This may need some tweaking still, but it's already pretty epic.

**Comment by nickolanack** [comment]
Cool, I agree. It is working for my own use case and will work for the solar system demo as is

**Pull request merged**